types/opt/value.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package opt
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
+// Value is an optional value to be JSON-encoded.
+// Unlike the specialized Bool type, an empty (unset) Value[T] is marshaled as
+// a JSON null regardless of the "omitempty" JSON tag option when using `encoding/json`.
+// With https://github.com/go-json-experiment/json, however, empty optional Values are
+// omitted from the output.
```

Comment:
Sorry, I don't quite understand why `omitempty` should not be used with `Value`. Can you please take a look at the `null_for_unset_omitempty_jsonv2` test?

Reply:
The problem is that it sometimes works, but has false positives.

Consider this example:
```go
var foo struct {
Map1 Value[map[string]string] `json:",omitempty"`
Map2 Value[map[string]string] `json:",omitzero"`
}
foo.Map1.Set(map[string]string{})
foo.Map2.Set(map[string]string{})

b, _ := jsonv2.Marshal(foo) // {"Map2":{}}
```

On the other hand, `omitzero` will always do the right thing since there is exactly one representation of an unset `Value`, which is the zero value itself.

Reply:
Got it, thanks! I updated the doc and the tests.